src/code/z_player_lib.c

```diff
@@ -2045,13 +2609,195 @@ void Player_DrawGetItem(PlayState* play, Player* player) {
     }
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80126AB4.s")
+void func_80126AB4(Player* player, Vec3f** arg1) {
+    if ((player->transformation == PLAYER_FORM_GORON) || (player->actor.id == ACTOR_EN_TEST3)) {
+        Math_Vec3f_Copy(&player->unk_AF0[1], &player->meleeWeaponInfo[0].base);
+        *arg1 = D_801C09B8;
+        return;
+    }
+
+    D_801C0994[1].x = D_801C0994[0].x;
+
+    if (player->unk_ADD >= 3) {
+        player->unk_ADD++;
+        D_801C0994[1].x *= 1.0f + ((9 - player->unk_ADD) * 0.1f);
+    }
+
+    D_801C0994[1].x += 1200.0f;
+    D_801C0994[2].x = D_801C0994[1].x;
+    *arg1 = D_801C0994;
+}
+
+void func_80126B8C(PlayState* play, Player* player) {
+    Vec3f* sp1C;
+
+    func_80126AB4(player, &sp1C);
+    func_8012669C(play, player, sp1C, D_801C0970);
+}
+
+// Zora boomerangs (?)
+void func_80126BD0(PlayState* play, Player* player, s32 arg2) {
+    if ((player->transformation != PLAYER_FORM_ZORA) || (player->rightHandType == PLAYER_MODELTYPE_RH_HOOKSHOT)) {
+        return;
+    }
+
+    if ((arg2 != 0) && (player->stateFlags1 & PLAYER_STATE1_400000)) {
+        OPEN_DISPS(play->state.gfxCtx);
+
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, object_link_zora_DL_0110A8);
+
+        CLOSE_DISPS(play->state.gfxCtx);
+    } else {
+        Actor* boomerangActor = player->boomerangActor;
```

Comment:
Reminder that we're going to have to rename this since dropped Deku Nuts also use it.